src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseStringContainsCharOverloadWithSingleCharacters.Fixer.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public sealed class UseStringContainsCharOverloadWithSingleCharactersCodeFix : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(
+            UseStringContainsCharOverloadWithSingleCharactersAnalyzer.CA1841);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var violatingNode = root.FindNode(context.Span, getInnermostNodeForTie: true);
+            if (context.Diagnostics[0].Properties.TryGetValue(UseStringContainsCharOverloadWithSingleCharactersAnalyzer.SourceStringPropertyKey, out var sourceString))
```

Comment:
Please add a defensive null check for `Properties`
```suggestion
            if (context.Diagnostics[0].Properties?.TryGetValue(UseStringContainsCharOverloadWithSingleCharactersAnalyzer.SourceStringPropertyKey, out var sourceString) == true)
```

Reply:
Had to go a little different route since there were changes to that section and the nullability analyzer was really not having it today with the exact change you proposed (honestly dont know why - it kept accusing that sourceString may be null even if i double checked it with a shortcircuit null check after the TryGet call) - hope this is ok now